### PR TITLE
Drop actor measurement metrics and introduce operation-related metrics

### PR DESCRIPTION
**Pull request by alenkacz**
Summary:
New metric names for metronome. I completely dropped the generic
Service and ActorMeasurement classes because I don't feel they
are bringing any value.

Instead I hand-picked metrics that make sense from my point of view.
If we agree on the names I will submit a documentation as another PR.

This is the current list of metrics introduced in this PR.

jobs.running.gauge
jobs.failed.counter
jobs.started.counter

debug.persistence.history.create.duration
debug.persistence.history.update.duration
debug.persistence.history.delete.duration

debug.persistence.job-run.create.duration
debug.persistence.job-run.update.duration
debug.persistence.job-run.delete.duration

debug.persistence.job-spec.create.duration
debug.persistence.job-spec.update.duration
debug.persistence.job-spec.delete.duration

debug.job-run.operations.list.duration
debug.job-run.operations.get.duration
debug.job-run.operations.kill.duration
debug.job-run.operations.list-active.duration
debug.job-run.operations.start.duration

debug.job-spec.operations.create.duration
debug.job-spec.operations.get.duration
debug.job-spec.operations.update.duration
debug.job-spec.operations.list.duration
debug.job-spec.operations.delete.duration

JIRA issues: DCOS-40729

### CODE REVIEW

File: jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceActor.scala [hunk 1]
```diff
@@ -77,6 +81,9 @@ class JobRunServiceActor(
   }
 
   def startJobRun(jobRun: JobRun): StartedJobRun = {
+    jobsRunningMetric.increment()
```
Comment (kensipe) [inline jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceActor.scala]: I'm wondering if this is misleading... the job at this point goes to the queue... it isn't running... and it is "started".  thoughts?

**Comment by kensipe** [comment]
this was a development issue... we will need to confirm we have the memory we need in prod.  I will add some docs for dev

File: jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunPersistenceActor.scala [hunk 2]
```diff
@@ -2,37 +2,41 @@ package dcos.metronome
...
-  repo:            Repository[JobRunId, JobRun],
-  val measurement: ServiceMeasurement) extends NoConcurrentRepoChange[JobRunId, JobRun, Unit] with ActorMeasurement {
+  id:      JobRunId,
+  repo:    Repository[JobRunId, JobRun],
+  metrics: Metrics) extends NoConcurrentRepoChange[JobRunId, JobRun, Unit] {
   import JobRunPersistenceActor._
   import context.dispatcher
 
-  override def receive: Receive = measure {
+  private val createJobRunTimeMetric = metrics.timer("debug.persistence.jobrun.create.duration")
```
Comment (ichernetsky) [inline jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunPersistenceActor.scala]: Should `jobrun` be replaced with `job-run` in metric names?

File: jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceActor.scala (hunk 1 above)
Comment (ichernetsky) [reply jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceActor.scala]: Should we just replace `running` to `active` in the metric name?

File: jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceDelegate.scala [hunk 3]
```diff
@@ -6,33 +6,41 @@ import akka.pattern.ask
...
   config:   JobRunConfig,
-  actorRef: ActorRef) extends JobRunService {
+  actorRef: ActorRef,
+  metrics:  Metrics) extends JobRunService {
 
   implicit val timeout: Timeout = config.askTimeout
   import JobRunServiceActor._
 
-  override def listRuns(filter: JobRun => Boolean): Future[Iterable[StartedJobRun]] = {
+  private val listRunsTimeMetric = metrics.timer("debug.jobrun.operations.list.duration")
```
Comment (ichernetsky) [inline jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceDelegate.scala]: `jobrun -> job-run`

File: jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecPersistenceActor.scala [hunk 4]
```diff
@@ -2,35 +2,39 @@ package dcos.metronome
...
-  repo:            Repository[JobId, JobSpec],
-  val measurement: ServiceMeasurement) extends NoConcurrentRepoChange[JobId, JobSpec, ActorRef] with ActorMeasurement {
+  id:      JobId,
+  repo:    Repository[JobId, JobSpec],
+  metrics: Metrics) extends NoConcurrentRepoChange[JobId, JobSpec, ActorRef] {
   import JobSpecPersistenceActor._
   import context.dispatcher
 
-  override def receive: Receive = measure {
+  private val createJobSpecTimeMetric = metrics.timer("debug.persistence.jobspec.create.duration")
```
Comment (ichernetsky) [inline jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecPersistenceActor.scala]: `jobspec -> job-spec`?

File: jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecServiceDelegate.scala [hunk 5]
```diff
@@ -7,32 +7,40 @@ import akka.util.Timeout
...
 
 import scala.concurrent.Future
 
 class JobSpecServiceDelegate(
   config:   JobSpecConfig,
-  actorRef: ActorRef) extends JobSpecService {
+  actorRef: ActorRef,
+  metrics:  Metrics) extends JobSpecService {
+
+  private val createJobSpecTimeMetric = metrics.timer("debug.jobspec.operations.create.duration")
```
Comment (ichernetsky) [inline jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecServiceDelegate.scala]: `jobspec -> job-spec`?

**Comment by kensipe** [comment]
@ichernetsky updated metric names as requested

**Comment by ichernetsky** [comment]
I edited the title and the description.

**Pull request merged**